### PR TITLE
tonic: fizz backport

### PR DESCRIPTION
**Pull request by loopfz**
Incorporates @wI2L (required for fizz integration: GetRouteByHandler getter) and @Rbeuque74 ("validate" tag) PRs.
Amended to keep backwards compatibility, and slight regression fix for inputs with slices/arrays in query.

### CODE REVIEW

**Comment by loopfz** [comment]
@wI2L @Rbeuque74 @nomonamo merging this tomorrow, do you see any issues ?

File: tonic/utils/swag/helpers.go [hunk 1]
```diff
@@ -6,12 +6,19 @@ import (
...
 	"github.com/loopfz/gadgeto/tonic/utils/swag/swagger"
 )
 
 const (
-	query_tag = "query"
-	path_tag  = "path"
+	queryTag         = tonic.TagQuery
+	pathTag          = tonic.TagPath
+	validateTag      = tonic.TagValidation
+	defaultTag       = "default"
```
Comment (wI2L) [inline tonic/utils/swag/helpers.go]: This is declared in `tonic`, maybe we should reuse it. `fiizz` will use this exported tag name from `tonic` to keep sync.

File: tonic/README [hunk 2]
```diff
@@ -9,13 +9,31 @@ of various parameters.
 Here is an example input object.
 
     type MyInput struct {
-        Foo int    `path:"foo,required"`
-        Bar float  `query:"bar,default=foobar"`
-        Baz string `json:"baz" binding:"required"`
+        Foo int    `path:"foo"`
+        Bar string `query:"bar" default="foobar"`
```
Comment (Rbeuque74) [inline tonic/README]: default:"foobar"

File: tonic/README [hunk 3]
```diff
@@ -9,13 +9,31 @@ of various parameters.
...
 
 Output objects can be of any type, and will be marshaled to JSON.
 
+Input validation is performed after binding into the object using the validator library
+(https://github.com/go-playground/validator). You can use the tag 'validate' on your object
+definition to perform validation inside the tonic handler phase.
+
+    type MyInput struct {
+        Foo int    `path:"foo" validate:"required,gt=10"`
+        Bar string `query:"bar" default="foobar" validate:"nefield=Baz"`
```
Comment (Rbeuque74) [inline tonic/README]: default:"foobar"

File: tonic/utils/swag/helpers.go (hunk 1 above)
Comment (Rbeuque74) [reply tonic/utils/swag/helpers.go]: Done, with some others. https://github.com/loopfz/gadgeto/pull/38/commits/e7868d461f68643decb52d53a0c07084ecd2fa2c

**Comment by loopfz** [comment]
LGTM

**Comment by wI2L** [comment]
LGTM

2018-03-12 23:31 GMT+01:00 Thomas Schaffer <notifications@github.com>:

**Pull request merged**